### PR TITLE
add functionality for getting events from data publishing

### PR DESCRIPTION
**Pull request by plavacquery**
receive event from data publishing

for getting events from idunn:
curl 'localhost:5000/v1/events?bbox=2.0667651,48.432533,2.9384989,49.0349191&raw_filter=bakery,*&size=5' |jq .

Do not forget to set kuzzle address and port

### CODE REVIEW

File: tests/test_kuzzle.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        assert firstEventData['address']['label'] == '30 Avenue Corentin Cariou, Paris'
+        assert firstEventData['address']['city'] == 'Paris'
+        assert firstEventData['blocks'][0]['type'] == 'event_opening_date'
+        assert firstEventData['blocks'][1]['type'] == 'event_description'
+        assert firstEventData['blocks'][2]['type'] == 'website'
+        assert firstEventData['blocks'][3]['type'] == 'images'
+        assert firstEventData['blocks'][0]['date_start'] == '2019-03-23T00:00:00.000Z'
+        assert firstEventData['blocks'][3]['images'] == [
+            {
+              "url": "https://s2.qwant.com/thumbr/0x0/b/8/ad1bd2ff50021ff6a1239585cc9ccde31e70072299c3cc910da54f9e791f7c/.jpg?u=&q=0&b=1&p=0&a=0",
```
Comment (amatissart) [inline tests/test_kuzzle.py]: this does not look correct. There is no image in the source document, this block should not be returned

File: tests/test_kuzzle.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import re
+import json
+import responses
+from .utils import override_settings
+from idunn.api.places_list import PlacesQueryParam
+
+from app import app, settings
+
+@pytest.fixture(scope="function")
+def limiter_test_normal():
```
Comment (amatissart) [inline tests/test_kuzzle.py]: this fixture should be renamed

File: tests/test_kuzzle.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        with responses.RequestsMock(assert_all_requests_are_fired=False) as rsps:
+            rsps.add('POST',
+                 re.compile(r'^http://localhost:7512/opendatasoft/events/'),
+                 status=200,
+                 json=json_event)
+
+
+            response = client.get(
+                url=f'http://localhost/v1/events?bbox=2.0667651,48.432533,2.9384989,49.0349191&raw_filter=bakery,*&size=5',
+            )
```
Comment (amatissart) [inline tests/test_kuzzle.py]: what is the expected status here ?

File: idunn/places/event.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    def build_address(self, lang):
+        """
+        Method to build the address field for an Address,
+        a Street, an Admin or a POI.
+        """
+
+        return {
+            "name": self.get('placename'),
+            "region": self.get('region'),
+            "department": self.get('department'),
```
[the remark this answers is not in the document]
Comment (plavacquery) [reply idunn/places/event.py]: do I remove them or I put it in label ? such as

"label": self.get('address') + ', ' + self.get('region') + ', ' + self.get('department'),

File: idunn/places/event.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            self['properties'] = {
+                p.get('key'): p.get('value') for p in self.get('properties', [])
+            }
+        self.properties = self['properties']
+
+    def get_local_name(self):
+        return self.get('title', '')
+
+    def get_id(self):
+        return self.get('uid')
```
[the remark this answers is not in the document]
Comment (plavacquery) [reply idunn/places/event.py]: Do not understand issue

File: idunn/api/places_list.py [hunk 6]
```diff
@@ -149,3 +153,72 @@ def get_places_bbox(bbox, es: Elasticsearch, indices: IndexNames, settings: Sett
...
+def get_events_bbox(bbox, query_params: http.QueryParams):
+    raw_params = get_raw_params(query_params)
+    kuzzle_address = settings.get(KUZZLE_CLUSTER_ADDRESS)
+    kuzzle_port = settings.get(KUZZLE_CLUSTER_PORT)
+
+    if not kuzzle_address or not kuzzle_port:
+        raise Exception(f"Missing kuzzle address or port: (port {KUZZLE_CLUSTER_PORT} is not set or address ${KUZZLE_CLUSTER_ADDRESS} is not set")
+
+    try:
+        params = PlacesQueryParam(**raw_params)
```
[the remark this answers is not in the document]
Comment (plavacquery) [reply idunn/api/places_list.py]: we should discuss about that

**Comment by jbgriesner** [comment]
I think all your `.idea` files are not 100% required by idunn ;)

File: idunn/blocks/events.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+    description = validators.String(allow_null=True)
+    free_text = validators.String(allow_null=True)
+    price = validators.String(allow_null=True)
+
+    @classmethod
+    def from_es(cls, es_poi, lang):
+        description = es_poi.get('description', None)
+        free_text =  es_poi.get('free_text', None)
+        price = es_poi.get('price', None)
```
Comment (amatissart) [inline idunn/blocks/events.py]: "princing_info" ?

File: idunn/places/event.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        return self.get('geo_loc')
+
+    def get_lang(self):
+        return self.get('lang')
+
+    def get_website(self):
+        return self.get('link')
+
+    def get_images_urls(self):
+        return [self.get('image_thumb', ''), self.get('image', '')]
```
Comment (amatissart) [inline idunn/places/event.py]: missing images should not be included in the list

File: tests/test_events_date.py [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+    )
+
+
+def test_event_day_complete():
+    """
+    We freeze time at 8:30 UTC (ie. 11:30 in Moscow)
+    The POI located in Moscow should be open since
+    it opens at 10:00 every day and the local time
+    is 11:30.
+    """
```
Comment (amatissart) [inline tests/test_events_date.py]: obsolete comments in this file

File: tests/test_events_informations.py [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+a POI located in a different timezone (Moscow) contains
+the correct information.
+
+For each test we freeze time at different datetimes and
+for each we create a fake different OpeningHourBlock from
+a raw json extracted from a POI located in Moscow city.
+
+TODO: a test that checks that opening hours can now span
+over midnight (branch 'new-parsing' of the repo)
+"""
```
Comment (amatissart) [inline tests/test_events_informations.py]: obsolete comments

File: tests/test_kuzzle.py (hunk 3 above)
Comment (amatissart) [reply tests/test_kuzzle.py]: check status 501 here

**Pull request merged**